scripts/Dockerfile

```diff
@@ -21,45 +21,44 @@ ENV CMAKE_C_COMPILER=/usr/bin/gcc
 ENV CMAKE_CXX_COMPILER=/usr/bin/g++
 ENV CMAKE_MAKE_PROGRAM=/usr/bin/make
 ENV SUNDIALS_INST=/home/pybamm/.local
-ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib:
+ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib
+
+RUN conda create -n pybamm python=3.11
+RUN conda init --all
+SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
+RUN conda install -y pip
 
 ARG IDAKLU
 ARG ODES
 ARG JAX
 ARG ALL
 
-RUN conda create -n pybamm python=3.9
-RUN conda init --all
-SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
-RUN conda install -y pip
+RUN pip install --upgrade --user pip setuptools wheel wget cmake
 
 RUN if [ "$IDAKLU" = "true" ]; then \
-    pip install --upgrade --user pip setuptools wheel wget && \
-    pip install cmake==3.22 && \
     python scripts/install_KLU_Sundials.py && \
     git clone https://github.com/pybind/pybind11.git && \
-    pip install --user -e ".[all,dev]"; \
+    pip install --user -e ".[all,dev,docs]"; \
     fi
 
 RUN if [ "$ODES" = "true" ]; then \
-    pip install cmake==3.22 && \
-    pip install --upgrade --user pip wget && \
     python scripts/install_KLU_Sundials.py && \
-    pip install --user -e ".[all,odes,dev]"; \
+    git clone https://github.com/pybind/pybind11.git && \
```

Comment:
Good catch, thanks. However, this would be valid only if both solvers are going to be exclusive of each other (see the other comment). Happy to remove otherwise